src/builder/QuarkBuilder.sol

```diff
@@ -231,7 +237,13 @@ contract QuarkBuilder {
                             chainAccountsList[i].chainId, transferIntent.chainId, transferIntent.assetSymbol
                         )
                 ) {
-                    aggregateTransferAssetAvailableBalance += Accounts.sumBalances(positions);
+                    // If the payment token is the transfer token and user opt for paying with the payment token, reduce the available balance by the maxCost
+                    if (payment.isToken && Strings.stringEqIgnoreCase(payment.currency, transferIntent.assetSymbol)) {
+                        uint256 maxCost = PaymentInfo.findMaxCost(payment, chainAccountsList[i].chainId);
+                        aggregateTransferAssetAvailableBalance += Accounts.sumBalances(positions) - maxCost;
+                    } else {
+                        aggregateTransferAssetAvailableBalance += Accounts.sumBalances(positions);
+                    }
```

Comment:
```suggestion
                    aggregateTransferAssetAvailableBalance += Accounts.sumBalances(positions);
                    // If the payment token is the transfer token and user opt for paying with the payment token, reduce the available balance by the maxCost
                    if (payment.isToken && Strings.stringEqIgnoreCase(payment.currency, transferIntent.assetSymbol)) {
                        uint256 maxCost = PaymentInfo.findMaxCost(payment, chainAccountsList[i].chainId);
                        aggregateTransferAssetAvailableBalance -= maxCost;
                    }
```

Reply:
nit suggestion: we don't need the `else` block if you keep L234 (`aggregateTransferAssetAvailableBalance += Accounts.sumBalances(positions);`). Feels cleaner this way to me personally.

Reply:
Sounds good.👍